tests/test_benchmark.py

```diff
@@ -423,6 +418,27 @@ def setUpClass(self):
     Building.COMPILER = CLANG
     Building.COMPILER_TEST_OPTS = [OPTIMIZATIONS]
 
+  # avoid depending on argument reception from the commandline
+  def hardcode_arguments(self, code):
+    if not code:
+      return code
+    main_pattern = 'int main(int argc, char **argv)'
+    assert main_pattern in code
+    code = code.replace(main_pattern, 'int benchmark_main(int argc, char **argv)')
+    code += '''
+      int main() {
+        // use volatiles so the compiler doesn't optimize all the work away
+        volatile int argc = 2;
+        typedef char** charStarStar;
+        volatile charStarStar argv;
+        argv[0] = (char*)"./program.exe";
+        argv[1] = (char*)"%s";
```

Comment:
Is there not some other way to inject argv at compile time?  Maybe `-s ARGV` or something like that?  This method seems pretty nasty.

Reply:
It is quite bad, yeah - but it works in a cross-compiler way, on emcc, native, and cheerp.

It's a horrible hack, but limited to the benchmark suite - that's the only reason I think it's acceptable.